### PR TITLE
fix: replacing course key with course number in enroll and unenroll email

### PR DESCRIPTION
**Pull request by MAbdurrehman12**
#### Pre-Flight checklist
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes https://github.com/mitodl/mitxonline/issues/279

#### What's this PR do?
This PR replaces the course key with course number (i.e: the last part of course.readable_id, after the final +) from the email templates of course enrollment and unenrollment.  

#### How should this be manually tested?
- Go to mitxonline
- Enroll into a course and then unenroll
- View the enrollment and unenrollment emails to verify that course key is now replaced with course number  

#### Screenshots (if appropriate)
Course enrollment email: 
enrollment email

Course enrollment email:
unenrollment email

### CODE REVIEW

File: courses/management/utils.py
Comment (MAbdurrehman12) [inline courses/management/utils.py:237]: @rhysyngsun  @pdpinch 
In order to get **course number** from **readable_id**, I have written a this custom function but this looks like a common use-case so what if we move this function to **ol-django -> common -> utils** and do it via opaque_keys, something like this: 
`from opaque_keys.edx.keys import CourseKey

course_key = CourseKey.from_string(enrollment.run.course.readable_id)
course_number = course_key.course
`
What do you suggest?

Comment (rhysyngsun) [reply courses/management/utils.py:237]: I think it probably justifies the addition of an edx-specific module in `ol-django` as opposed to putting it in `common`

Comment (pdpinch) [reply courses/management/utils.py:237]: Thanks @MAbdurrehman12. I think @rhysyngsun makes a good suggestion.

Comment (MAbdurrehman12) [reply courses/management/utils.py:237]: Noted.
Thank you

**Comment by pdpinch** [comment]
Screenshots look good

**Comment by MAbdurrehman12** [comment]
A custom function **get_course_number** was created in this PR, to fetch the course_number from readable_id. Since this function seems to be quite common hence we decided to move this to ol-django so it can be reused in future, when needed.
For this, a PR has been created in ol-django: https://github.com/mitodl/ol-django/pull/48

Currently, this PR is being blocked by ol-django PR, so as soon as the ol-django PR is merged, upon its new release, we can update this PR to import the function from ol-django.

**Comment by pdpinch** [comment]
What's the status of this?

**Comment by MAbdurrehman12** [comment]
> What's the status of this?

The required util was merged in ol-django in this [PR](https://github.com/mitodl/ol-django/pull/48), though couldn't find it's release so I'll get in touch with Nathan regarding ol-django release. 
Once it is released, I'll update this PR.

File: courses/templates/mail/course_run_enrollment/body.html
Comment (arslanashraf7) [inline courses/templates/mail/course_run_enrollment/body.html:12]: As per my above comment:

```suggestion
                        You have been enrolled in {{ enrollment.run.course_number }} {{ enrollment.run.course.title }}. The course should now appear on your <a href="{{ base_url }}{% url 'user-dashboard' %}">{{ site_name }} dashboard</a>.
```

**Review by arslanashraf7 — CHANGES_REQUESTED** [review]
I have requested some changes, let me know if you have any ambiguity about them.

File: courses/mail_api.py
Comment (MAbdurrehman12) [reply courses/mail_api.py:28]: That's a good suggestion.
Thanks!

File: courses/models.py
Comment (arslanashraf7) [inline courses/models.py:436]:
```suggestion
        """
        Returns:
            str: Course number from the readable_id
        """
```

**Review by arslanashraf7 — APPROVED** [review]
Looks great 👍 , Just one minor comment, feel free to merge after that.

I know you know it already but remember to squash the commits & merge.

File: courses/models.py
Comment (MAbdurrehman12) [reply courses/models.py]: "Course number from the readable_id": I guess this needs a little explanation as well, as the course number is fetched from courseware_id which is the last part of readable_id, after the final +.
Anyhow, I have updated the description.

**Comment by briangrossman** [comment]
@MAbdurrehman12 - I just tested this on RC. It works great! 👍

**Comment by MAbdurrehman12** [comment]
> @MAbdurrehman12 - I just tested this on RC. It works great! 👍

Thanks Brian!

**Pull request merged**
